warehouse/templates/manage/project/settings.html

```diff
@@ -265,7 +265,13 @@ <h3>{% trans %}Delete project{% endtrans %}</h3>
         </small>
       </li>
     </ul>
+    <ul class="no-bullets">
+      <li><input type="checkbox" data-action="input->delete-confirm#check" data-delete-confirm-target="input"> I understand that I will not be able to re-upload any deleted versions.</li>
+      <li><input type="checkbox" data-action="input->delete-confirm#check" data-delete-confirm-target="input"> I understand that I am permanently deleting all releases for this project.</li>
+      <li><input type="checkbox" data-action="input->delete-confirm#check" data-delete-confirm-target="input"> I understand that I am releasing this project name for use by any other PyPI user.</li>
```

Comment:
Maybe add a "I understand that I may not be able to re-register the project name under some circumstances" and link to https://pypi.org/help/#project-name